<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindowClass</class>
 <widget class="QMainWindow" name="MainWindowClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1025</width>
    <height>709</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>QtTest</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QPushButton" name="genRndFaceBtn">
    <property name="geometry">
     <rect>
      <x>580</x>
      <y>80</y>
      <width>421</width>
      <height>28</height>
     </rect>
    </property>
    <property name="text">
     <string>generate random face</string>
    </property>
   </widget>
   <widget class="OpenGLWidget" name="openGLWidget">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>20</y>
      <width>531</width>
      <height>651</height>
     </rect>
    </property>
   </widget>
   <widget class="QWidget" name="horizontalLayoutWidget">
    <property name="geometry">
     <rect>
      <x>580</x>
      <y>30</y>
      <width>421</width>
      <height>41</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="rndRangeLayout">
     <item>
      <widget class="QLabel" name="rndRangeLabel">
       <property name="text">
        <string>random range</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_5">
       <property name="text">
        <string>0.0</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSlider" name="rndRangeSlider">
       <property name="minimum">
        <number>0</number>
       </property>
       <property name="maximum">
        <number>200</number>
       </property>
       <property name="pageStep">
        <number>2</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_6">
       <property name="text">
        <string>2.0</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="horizontalLayoutWidget_2">
    <property name="geometry">
     <rect>
      <x>580</x>
      <y>140</y>
      <width>421</width>
      <height>41</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QLabel" name="label">
       <property name="text">
        <string>No.</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="shapeComboBox"/>
     </item>
     <item>
      <widget class="QLabel" name="label_7">
       <property name="text">
        <string>-10.0</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSlider" name="shapeSlider">
       <property name="minimum">
        <number>-1000</number>
       </property>
       <property name="maximum">
        <number>1000</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_8">
       <property name="text">
        <string>10.0</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QPushButton" name="setShapePcBtn">
    <property name="geometry">
     <rect>
      <x>580</x>
      <y>190</y>
      <width>421</width>
      <height>28</height>
     </rect>
    </property>
    <property name="text">
     <string>set shape PC</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>580</x>
      <y>120</y>
      <width>421</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>=================== shape PC =======================</string>
    </property>
   </widget>
   <widget class="QPushButton" name="setTexPcBtn">
    <property name="geometry">
     <rect>
      <x>580</x>
      <y>300</y>
      <width>421</width>
      <height>28</height>
     </rect>
    </property>
    <property name="text">
     <string>set texture PC</string>
    </property>
   </widget>
   <widget class="QWidget" name="horizontalLayoutWidget_3">
    <property name="geometry">
     <rect>
      <x>580</x>
      <y>250</y>
      <width>421</width>
      <height>41</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <item>
      <widget class="QLabel" name="label_3">
       <property name="text">
        <string>No.</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="texComboBox"/>
     </item>
     <item>
      <widget class="QLabel" name="label_9">
       <property name="text">
        <string>-10.0</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSlider" name="texSlider">
       <property name="minimum">
        <number>-1000</number>
       </property>
       <property name="maximum">
        <number>1000</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_10">
       <property name="text">
        <string>10.0</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>580</x>
      <y>230</y>
      <width>421</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>=================== texture PC =======================</string>
    </property>
   </widget>
   <widget class="QWidget" name="horizontalLayoutWidget_4">
    <property name="geometry">
     <rect>
      <x>580</x>
      <y>620</y>
      <width>421</width>
      <height>31</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_3">
     <property name="sizeConstraint">
      <enum>QLayout::SetFixedSize</enum>
     </property>
     <item>
      <widget class="QLabel" name="label_11">
       <property name="text">
        <string>filename:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QTextEdit" name="saveFilename">
       <property name="maximumSize">
        <size>
         <width>236</width>
         <height>29</height>
        </size>
       </property>
       <property name="html">
        <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'SimSun'; font-size:9pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'SimSun';&quot;&gt;rnd_face.ply&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="savePlyBtn">
       <property name="text">
        <string>save as PLY</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1025</width>
     <height>26</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>文件</string>
    </property>
    <addaction name="actionload"/>
    <addaction name="actionchange_model"/>
   </widget>
   <widget class="QMenu" name="menu_2">
    <property name="title">
     <string>帮助</string>
    </property>
    <addaction name="actionBFM"/>
   </widget>
   <addaction name="menu"/>
   <addaction name="menu_2"/>
  </widget>
  <action name="actionload">
   <property name="text">
    <string>载入模型</string>
   </property>
  </action>
  <action name="actionchange_model">
   <property name="text">
    <string>更换模型</string>
   </property>
  </action>
  <action name="actionBFM">
   <property name="text">
    <string>BFM模型预处理</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>OpenGLWidget</class>
   <extends>QOpenGLWidget</extends>
   <header location="global">OpenGLWidget.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="MainWindow.qrc"/>
 </resources>
 <connections/>
</ui>
